src/libtls/tls_crypto.c

```diff
@@ -2322,6 +2380,36 @@ int tls_crypto_get_supported_suites(bool null, tls_version_t version,
 	return count;
 }
 
+/**
+ * See header.
+ */
+int tls_crypto_get_supported_groups(diffie_hellman_group_t **out)
+{
+	enumerator_t *enumerator;
+	diffie_hellman_group_t groups[countof(curves)];
+	diffie_hellman_group_t group;
+	tls_named_group_t curve;
+	int count = 0, i;
+
+	enumerator = get_filtered_ec_enumerator();
```

Comment:
Actually, here only `group_filter` around the DH implementations is necessary (i.e. like the original `create_ec_enumerator()`. The whole point of splitting the filter was that this enumeration here will not be affected by the config (e.g. setting the global `libtls.curves` should not affect with which DH groups the unit tests are run).

Reply:
Ah, I see... I think I now understood it. Thank you for this elaboration!